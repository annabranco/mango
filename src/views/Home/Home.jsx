import React from "react";
import { HomeLink, HomeText, HomeWrapper } from "./Home.styles";

const Home = () => {
  return (
    <HomeWrapper>
      <HomeText>Please select exercise:</HomeText>
      <HomeLink to="exercise1">Exercise 1</HomeLink>
      <HomeLink to="exercise2">Exercise 2</HomeLink>
    </HomeWrapper>
  );
};

export default Home;
