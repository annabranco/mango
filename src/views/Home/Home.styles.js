import styled from "styled-components";
import { Link } from "react-router-dom";

export const HomeLink = styled(Link)`
  margin: 20px auto;
  font-size: 2rem;
  font-weight: 900;
  color: green;
  -webkit-text-stroke: 1px black;
  text-shadow: 1px 1px 10px rgba(0, 0, 0, 0.5);
  transition: all 0.2s ease;

  &:hover {
    filter: saturate(2) brightness(2);
  }
`;
HomeLink.displayName = "HomeLink";

export const HomeText = styled.p`
  margin-bottom: 50px;
  font-size: 1.15rem;
  font-weight: 400;
  font-style: italic;
`;
HomeText.displayName = "HomeText";

export const HomeWrapper = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 50px;
`;
HomeWrapper.displayName = "HomeWrapper";
