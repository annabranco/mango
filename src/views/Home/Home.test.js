import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";

import Home from "./Home";

configure({ adapter: new Adapter() });

describe("[Home] component with normal values", () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<Home />);
  });

  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
