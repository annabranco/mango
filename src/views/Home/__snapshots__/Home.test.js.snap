// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Home] component with normal values should match the snapshot 1`] = `
<HomeWrapper>
  <HomeText>
    Please select exercise:
  </HomeText>
  <HomeLink
    to="exercise1"
  >
    Exercise 1
  </HomeLink>
  <HomeLink
    to="exercise2"
  >
    Exercise 2
  </HomeLink>
</HomeWrapper>
`;
